aviatrix/resource_aviatrix_remote_syslog.go

```diff
@@ -0,0 +1,161 @@
+package aviatrix
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aviatrix/goaviatrix"
+)
+
+func resourceAviatrixRemoteSyslog() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAviatrixRemoteSyslogCreate,
+		Read:   resourceAviatrixRemoteSyslogRead,
+		Update: resourceAviatrixRemoteSyslogUpdate,
+		Delete: resourceAviatrixRemoteSyslogDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"index": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      0,
+				ValidateFunc: validation.IntBetween(0, 9),
+				Description:  "Profile index: a total of 10 profiles from index 0 to 9 are supported for remote syslog, while index 9 is reserved for CoPilot.",
+			},
+			"server": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Server: FQDN or IP address of the remote syslog server",
+			},
+			"port": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Port: Listening port of the remote syslog server (6514 by default)",
+			},
+			"ca_certificate_file_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "CA Certificate: Certificate Authority (CA) certificate",
+			},
+			"public_certificate_file_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Server Public Certificate: Public certificate of the controller signed by the same CA",
+			},
+			"private_key_file_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Server Private Key: Private key of the controller that pairs with the public certificate",
+			},
+			"protocol": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "TCP",
+				ValidateFunc: validation.StringInSlice([]string{"TCP", "UDP"}, false),
+				Description:  "Protocol: TCP or UDP (TCP by default)",
+			},
+			"template": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Optional Custom Template: Useful when forwarding to 3rd party servers like Datadog or Sumo",
+			},
+			"exclude_gateway": {
```

Comment:
Since this is a collection of items the name should be plural. Please use something like `excluded_gateways`